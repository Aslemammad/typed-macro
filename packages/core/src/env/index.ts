export { EnvContext, Modules } from './env'
