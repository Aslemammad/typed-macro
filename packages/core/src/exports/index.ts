export { Exportable, ModularizedExportable } from './exports'
