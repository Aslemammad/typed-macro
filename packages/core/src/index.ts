export * from './env'
export * from './macro'
export * from './exports'
export * from './provider'
export * from '../types'
