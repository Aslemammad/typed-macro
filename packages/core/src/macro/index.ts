export {
  Macro,
  MacroHandler,
  MacroCall,
  isMacro,
  YieldTask,
  YieldUnit,
  SuspendableTransform,
} from './macro'
export { MacroContext, MacroTransformContext } from './context'
export { ImportOption, MacroHelper } from './helper'
export { Babel } from './babel'
export { State } from './state'
export { defineMacro, MacroBuilder } from './builder'
