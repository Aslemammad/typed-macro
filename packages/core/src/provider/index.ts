export { MacroProviderOptions } from './options'
export { MacroProviderHooks } from './hooks'
export {
  MacroProvider,
  isMacroProvider,
  InstantiatedMacroProvider,
} from './provider'
export { defineMacroProvider } from './builder'
