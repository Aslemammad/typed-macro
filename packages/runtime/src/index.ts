export * from './transformer'
export * from './normalizer'
export * from './renderer'
export * from './runtime'
export * from './loader'
export * from './filter'
export * from './hook'
