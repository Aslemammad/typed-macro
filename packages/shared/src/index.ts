export * from './utils'
export * from './version'
export * from './types'
