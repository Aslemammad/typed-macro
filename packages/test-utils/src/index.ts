export * from './transformer'
export * from './runtime'
export * from './renderer'
export * from './env'
