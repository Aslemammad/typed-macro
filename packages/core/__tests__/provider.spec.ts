import { defineMacroProvider, isMacroProvider } from '../src'

describe('function defineMacroProvider()', () => {
  it('should work', () => {
    // currently defineMacroProvider() is just an entry for macroProvider()
    const createdWithLiteral = defineMacroProvider({
      id: 'test',
      exports: {},
    })
    const createdWithFactory = defineMacroProvider(() => {
      return { id: 'test', exports: {} }
    })
    expect(createdWithFactory).toMatchSnapshot()
    expect(createdWithLiteral).toMatchSnapshot()
  })
})

describe('function isMacroProvider()', () => {
  it('should work', () => {
    const createdWithLiteral = defineMacroProvider({
      id: 'test',
      exports: {},
    })
    const createdWithFactory = defineMacroProvider(() => {
      return { id: 'test', exports: {} }
    })
    expect(isMacroProvider(createdWithFactory)).toBe(true)
    expect(isMacroProvider(createdWithLiteral)).toBe(true)
  })
})
