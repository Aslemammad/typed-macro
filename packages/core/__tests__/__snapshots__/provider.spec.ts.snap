// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function defineMacroProvider() should work 1`] = `[Function]`;

exports[`function defineMacroProvider() should work 2`] = `
Object {
  "__is_macro_provider": true,
  "exports": Object {},
  "id": "test",
  "version": "1",
}
`;
